### PR TITLE
Running AutoRest Linter Rules in travis-ci

### PR DESCRIPTION
**Pull request by vishrutshah**
@amarzavery @veronicagg @sarangan12 

Here is the dry-run on how we want to run linter rules inside travis-ci. Please review it and let me know your thoughts. 

Sample Outputs: https://travis-ci.org/vishrutshah/azure-rest-api-specs/builds/197485389

Currently it runs on all the swagger specs, once this is done we can accomodate linter.js to run on only changed PR files as well.

### CODE REVIEW

File: test/linter.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  return isWindows ? cmd : ('mono ' + cmd);
+};
+
+describe('AutoRest Linter validation:', function () {
+  var autoRestLocation = './AutoRest.*/tools/AutoRest.exe';
+  _(swaggers).each(function(swagger) {
+    it(swagger + ' should follow linter rules.', function(done) {
+      var cmd = clrCmd(autoRestLocation + ' -CodeGenerator None -I ' + swagger);
+      console.log(cmd);
+      exec(cmd, function(error, stdout, stderr) {
```
Comment (amarzavery) [inline test/linter.js]: Were you able to solve the problem to get everything in sequence?

File: package.json [hunk 2]
```diff
@@ -29,6 +29,6 @@
     "url": "http://github.com/azure/azure-rest-api-specs/issues"
   },
   "scripts": {
-    "test": "mocha -t 5000"
+    "test": "mocha -t 500000"
```
Comment (amarzavery) [inline package.json]: What was the need to bump up the value to such a high number?

Comment (vishrutshah) [inline package.json]: network spec took time, so i did bump up with 2 zeros.

File: test/linter.js (hunk 1 above)
Comment (vishrutshah) [inline test/linter.js]: Yes actually when i added `-CodeGenerator None` in command then it took less time and things started to fall in place without using `execSync`

**Review by amarzavery — APPROVED** [review]

**Pull request merged**